ssl/quic/quic_channel.c

```diff
@@ -2220,6 +2220,14 @@ static void ch_rx_handle_packet(QUIC_CHANNEL *ch)
              */
             return;
 
+        /*
+         * RFC 9000 s 17.2.5.2: After the client has received and processed an
+         * Initial or Retry packet from the server, it MUST discard any
+         * subsequent Retry packets that it receives.
+         */
+        if ((ch->el_discarded & (1U << QUIC_ENC_LEVEL_INITIAL)) != 0)
```

Comment:
Use `ch->have_received_enc_pkt` instead

This will not handle cases where we have received an Initial packet but not dropped the Initial EL yet